extension/scripts/features/faction-member-filter/ttFactionMemberFilter.js

```diff
@@ -0,0 +1,279 @@
+		{
+			storage: ["settings.pages.faction.memberFilter"],
+		},
+		null
+	);
+
+	function addListener() {
+		if (isOwnFaction()) {
+			CUSTOM_LISTENERS[EVENT_CHANNELS.FACTION_INFO].push(() => {
+				if (!feature.enabled()) return;
+
+				addFilter();
+			});
+		}
+	}
+
+	const localFilters = {};
+	async function addFilter() {
+		await requireElement("#faction-info-members .members-list .table-row");
+
+		const { content } = createContainer("Member Filter", {
+			class: "mt10",
+			nextElement: document.getElementById("faction-info-members"),
+			compact: true,
+			filter: true,
+		});
+
+		const statistics = createStatistics();
+		content.appendChild(statistics.element);
+		localFilters["Statistics"] = { updateStatistics: statistics.updateStatistics };
+
+		const filterContent = document.newElement({
+			type: "div",
+			class: "content",
+		});
+
+		const activityFilter = createFilterSection({
+			type: "Activity",
+			defaults: filters.faction.activity,
+			callback: applyFilter,
+		});
+		filterContent.appendChild(activityFilter.element);
+		localFilters["Activity"] = { getSelections: activityFilter.getSelections };
+
+		const specialFilter = createFilterSection({
+			title: "Special",
+			ynCheckboxes: ["Fedded", "New Player", "In Company", "Is Donator"],
+			defaults: filters.faction.special,
+			callback: applyFilter,
+		});
+		filterContent.appendChild(specialFilter.element);
+		localFilters["Special"] = { getSelections: specialFilter.getSelections };
+
+		const newDiv = document.newElement({ type: "div" });
+
+		const positionFilter = createFilterSection({
+			title: "Position",
+			select: getPositions(),
+			defaults: "",
+			callback: applyFilter,
+		});
+		newDiv.appendChild(positionFilter.element);
+		localFilters["Position"] = { getSelected: positionFilter.getSelected };
+
+		const statusFilter = createFilterSection({
+			title: "Status",
+			select: [
+				{ value: "", description: "All" },
+				{ value: "------", description: "------", disabled: true },
+				{ value: "Okay" , description: "Okay" },
+				{ value: "Hospital" , description: "Hospital" },
+				{ value: "Jail" , description: "Jail" },
+				{ value: "Traveling" , description: "Traveling" },
+			],
+			defaults: "",
+			callback: applyFilter,
+		});
+		newDiv.appendChild(statusFilter.element);
+		localFilters["Status"] = { getSelected: statusFilter.getSelected };
+
+		filterContent.appendChild(newDiv);
+
+		const levelFilter = createFilterSection({
+			title: "Level Filter",
+			noTitle: true,
+			slider: {
+				min: 1,
+				max: 100,
+				step: 1,
+				valueLow: filters.faction.levelStart,
+				valueHigh: filters.faction.levelEnd,
+			},
+			callback: applyFilter,
+		});
+		filterContent.appendChild(levelFilter.element);
+		localFilters["Level Filter"] = { getStartEnd: levelFilter.getStartEnd, updateCounter: levelFilter.updateCounter };
+
+		if (settings.scripts.lastAction.factionMember) {
+			await requireElement(".members-list .table-body.tt-modified");
```

Comment:
This should not be the way it's done. Try using custom listeners so indicate it and trigger updates etc. While it would probably be fine for this section, it wouldn't be for stat estimates, which will get added later possibly.